Tropos/Controllers/SettingsController.swift

```diff
@@ -0,0 +1,40 @@
+import Foundation
+
+@objc(TRSettingsController) final class SettingsController: NSObject {
+    private let locale: NSLocale
+
+    var unitSystem: UnitSystem {
+        get {
+            let rawUnitSystem = NSUserDefaults.standardUserDefaults().integerForKey(TRSettingsUnitSystemKey)
+            return UnitSystem(rawValue: rawUnitSystem)!
+        }
+        set {
+            NSUserDefaults.standardUserDefaults().setInteger(newValue.rawValue, forKey: TRSettingsUnitSystemKey)
+        }
+    }
+
+    init(locale: NSLocale) {
+        self.locale = locale
+    }
+
+    convenience override init() {
```

Comment:
If I remove the convenience initialiser, I get this runtime crash:

```
Tropos/Controllers/SettingsController.swift: 3: 41: fatal error: use of unimplemented initializer 'init()' for class 'Tropos.SettingsController'
```

It seems that Swift doesn't synthesize the default initialiser for you, and implementing your own designated initialiser removes the inherited `-[NSObject init]`.